embedded-graphics/MIGRATING-0.5-0.6.md

```diff
@@ -0,0 +1,422 @@
++
++ image.draw(&mut display)?;
+```
+
+## For driver authors
+
+Adding embedded-graphics support to a display driver has changed somewhat. The `Drawing` trait has been replaced by the `DrawTarget` trait which provides a slightly different interface.
+
+An associated error type is also added to `DrawTarget` to allow for better error handling than a `panic!()`.
+
+Below is a reduced example taken from the [ssd1306](https://crates.io/crates/ssd1306) driver. It uses `BinaryColor` as pixels on the SSD1306 can only be on or off.
+
+```rust
+use embedded_graphics::{
+    drawable,
+    geometry::Size,
+    pixelcolor::{
+        raw::{RawData, RawU1},
+        BinaryColor, PixelColor,
+    },
+    DrawTarget,
+};
+
+impl<C> DrawTarget<C> for DisplayDriver
+where
+    C: PixelColor + Into<BinaryColor>,
+{
+    type Error = DI::Error;
+
+    fn draw_pixel(&mut self, pixel: drawable::Pixel<C>) -> Result<(), Self::Error> {
+        let drawable::Pixel(pos, color) = pixel;
+
+        // Guard against negative values. All positive i32 values from `pos` can be represented in
+        // the `u32`s that `set_pixel()` accepts...
+        if pos.x < 0 || pos.y < 0 {
+            return Ok(());
+        }
+
+        // ... which makes the `as` coercions here safe.
+        self.set_pixel(
+            pos.x as u32,
+            pos.y as u32,
+            // Convery color to BinaryColor, then into underlying raw storage u8
+            color.into().into_storage(),
+        );
+
+        Ok(())
+    }
+
+    fn size(&self) -> Size {
+        let (w, h) = self.get_dimensions();
+
+        Size::new(w as u32, h as u32)
+    }
+}
+```
+
+Some notes on the above:
+
+- The `Drawing` trait is renamed to `DrawTarget`.
+- This implementation takes a `C: PixelColor + Into<BinaryColor>` allowing items using other color types to be used on the monochrome display. If this is not possible or undesired, use the appropriate pixel type directly. For example:
+
+  ```rust
+  impl DrawTarget<Rgb565> for DisplayDriver {
+      // ...
+  }
+  ```
+
+- The `draw()` method is replaced by `draw_pixel()`. This method should handle setting of individual pixels on the display. How it does this is at the descretion of the display driver (pixel buffer, immediate mode, etc).
+- A new `size()` method is now a required item. This should return the width and height of the display as a `Size`.
+- The implementation of `DrawTarget` must now provide an associated `Error` type. In the above example `core::convert::Infallible` is used, however a better error type should be used to communicate hardware failures, etc to the user.
+- Any pixels that are offscreen (negative coordinates or greater than display dimensions) should result in a noop. In the example above, `self.set_pixel()` (defined by the SSD1306 driver) will not attempt to set any pixels beyond the positive screen limits.
+- `draw_pixel()` now returns `Result<(), Self::Error>` to account for driver error handling.
```

Comment:
The other `draw_...` methods and the reason for their existence  should also be mentioned.